packages/react/src/primitives/SearchField/useSearchField.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+
+import { isFunction } from '../shared/utils';
+import { SearchFieldProps } from '../types';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
+
+const ESCAPE_KEY = 'Escape';
+const ENTER_KEY = 'Enter';
+const DEFAULT_KEYS = [ESCAPE_KEY, ENTER_KEY];
+
+export const useSearchField = ({
+  onSubmit,
+  onClear,
+  externalRef,
+}: Partial<
+  SearchFieldProps & { externalRef: React.ForwardedRef<HTMLInputElement> }
+>) => {
+  const [value, setValue] = React.useState<string>('');
```

Comment:
do we need to set it as an empty string? I wonder if `undefined` or `null` is a better default value here.

Reply:
I believe `undefined` will result in the classic React error "input changed from controlled to uncontrolled", so it's better to make it an empty string.

Reply:
👍🏼